### PR TITLE
Correct a few nullable annotations

### PR DESCRIPTION
**Pull request by geeknoid**
This matches the other TryParse signatures in the same assembly to reflect the reality of potentially returning null on error.

### CODE REVIEW

**Comment by geeknoid** [comment]
@davidfowl

File: src/Http/Headers/src/EntityTagHeaderValue.cs [hunk 1]
```diff
@@ -159,10 +159,10 @@ public static EntityTagHeaderValue Parse(StringSegment input)
     /// <param name="input">The value to parse.</param>
     /// <param name="parsedValue">The parsed value.</param>
     /// <returns><see langword="true"/> if input is a valid <see cref="EntityTagHeaderValue"/>, otherwise <see langword="false"/>.</returns>
-    public static bool TryParse(StringSegment input, [NotNullWhen(true)] out EntityTagHeaderValue parsedValue)
+    public static bool TryParse(StringSegment input, [NotNullWhen(true)] out EntityTagHeaderValue? parsedValue)
```
[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Http/Headers/src/EntityTagHeaderValue.cs:162]: No, the ResponseCaching code declares a non-nullable variable and passes it in. You just need to update the variable to be nullable.

[the remark this answers is not in the document]
Comment (geeknoid) [reply src/Http/Headers/src/EntityTagHeaderValue.cs:162]: Yep, just needed to add a ?

**Comment by BrennanConroy** [comment]
Real build errors. Likely similar fix that went into response caching needed.

**Comment by geeknoid** [comment]
I sprinkled some more ? around...

File: src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs [hunk 2]
```diff
@@ -438,7 +438,7 @@ internal static bool ContentIsNotModified(ResponseCachingContext context)
                 return true;
             }
 
-            EntityTagHeaderValue eTag;
+            EntityTagHeaderValue?    eTag;
```
Comment (JamesNK) [inline src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs:441]:
```suggestion
            EntityTagHeaderValue? eTag;
```

**Review by JamesNK — APPROVED** [review]

**Pull request merged**